### PR TITLE
add hybrid version with cffi

### PR DESCRIPTION
**Pull request by JeroenKnoops**
Add hybrid application with cffi.

Two disadvantages of this implementation:

1. I have to specify the `dylib` file, on windows it's `ddl`.. so implementation is mac osx only.
2. `libpi_monte_carlo_ffi.dylib` is >2.1 mb... so this is sub-optimal... added new experiment suggestion (in README) to compile to WASM

### CODE REVIEW

File: hybrid/pi-monte-carlo.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import random as r
+#import math as m
```
Comment (timovandeput) [inline hybrid/pi-monte-carlo.py]: This looks like a line that is no longer needed?

File: hybrid/pi-monte-carlo/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        let y: f64 = rng.gen::<f64>();
+        let p: f64 = x * x + y * y;
+
+        if p < 1.0 {
+            count += 1;
+        }
+    }
+
+    let pi_est: f64 = (count as f64) * 4.0 / (trials as f64);
+    pi_est
```
Comment (timovandeput) [inline hybrid/pi-monte-carlo/src/lib.rs]: Is it idiomatic in Rust to declare the return value as a local variable first? (The type is already established in the method signature.)

File: hybrid/pi-monte-carlo/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+#[no_mangle]
+pub fn montecarlo_pi(trials: u32) -> f64 {
+    let mut count: u32 = 0;
+    let mut rng = rand::thread_rng();
+
+    for _i in 1..trials {
+        let x: f64 = rng.gen::<f64>();
+        let y: f64 = rng.gen::<f64>();
+        let p: f64 = x * x + y * y;
```
Comment (timovandeput) [inline hybrid/pi-monte-carlo/src/lib.rs]: I personally don't like such cryptic names. :-) 

What is a "p"? (Looks like it refers to the distance to the center of the dart board.)

**Review by timovandeput — APPROVED** [review]

**Comment by JeroenKnoops** [comment]
Thanks, I will work on this.

File: hybrid/pi-monte-carlo.py (hunk 1 above)
Comment (JeroenKnoops) [reply hybrid/pi-monte-carlo.py]: Done!

File: hybrid/pi-monte-carlo/src/lib.rs (hunk 2 above)
Comment (JeroenKnoops) [reply hybrid/pi-monte-carlo/src/lib.rs]: Done

File: hybrid/pi-monte-carlo/src/lib.rs (hunk 3 above)
Comment (JeroenKnoops) [reply hybrid/pi-monte-carlo/src/lib.rs]: Done

**Review by timovandeput — APPROVED** [review]

**Pull request merged**